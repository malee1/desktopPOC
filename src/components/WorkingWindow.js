import { React, useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { useDispatch } from 'react-redux';
import axios from 'axios';
import AppBar from './AppBar';
import IFrame from './IFrame';
import { getUrlsApi } from '../constants/api';
import { getUrls, setIFrame } from '../redux/actions';

const useStyles = makeStyles({
  root: {
    height: '80vh',
    display: 'flex',
    padding: '5px'
  },
  appBarContainer: {
    minWidth: '15vw',
    backgroundColor: 'lightGrey'
  },
  iFrameContainer: {
    flexGrow: 16
  }
});

export default function WorkingWindow() {
  const classes = useStyles();

  const url = getUrlsApi;

  const dispatch = useDispatch();

  useEffect(() => {
    const getUrlData = async () => {
      try {
        const response = await axios(url);
        // console.log(response.data);
        dispatch(getUrls(response.data));
        // console.log(response.data.urlOne.url);
        dispatch(setIFrame(response.data.urlOne.url));
      } catch (error) {
        console.log(`Error getting url data: ${error}`);
      }
    };
    getUrlData();
  }, [url, dispatch]);

  return (
    <div className={classes.root}>
      <div className={classes.appBarContainer}>
        <AppBar />
      </div>
      <div className={classes.iFrameContainer}>
        <IFrame />
      </div>
    </div>
  );
}
