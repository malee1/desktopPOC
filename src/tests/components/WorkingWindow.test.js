import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';

import WorkingWindow from '../../components/WorkingWindow';

const mockStore = configureMockStore();
const store = mockStore({
  user: {
    data: {
      fileId: '123456'
    }
  },
  urls: {
    urls: {
      urlOne: {
        name: 'test-1',
        url: 'test-url-1'
      },
      urlTwo: {
        name: 'test-2',
        url: 'test-url-2'
      },
      urlThree: {
        name: 'test-3',
        url: 'test-url-3'
      }
    }
  },
  iFrame: {
    iFrameUrl: 'test-url'
  }
});

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

describe('Panel renders correctly', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <WorkingWindow />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders one component', () => {
    const { getAllByTestId } = render(
      <Provider store={store}>
        <WorkingWindow />
      </Provider>
    );
    const panel = getAllByTestId('working-window');
    expect(panel.length).toBe(1);
  });
});
