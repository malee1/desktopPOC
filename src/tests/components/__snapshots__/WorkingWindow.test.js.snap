// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly matches the snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="working-window"
>
  <div
    className="makeStyles-appBarContainer-2"
  >
    <div
      className="makeStyles-root-5"
      data-testid="app-bar"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5"
      >
        App Bar
      </h5>
      <div
        className="MuiButtonGroup-root makeStyles-buttonGroup-6 MuiButtonGroup-vertical"
        role="group"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlinedPrimary makeStyles-button-7 MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            test-1
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlinedPrimary makeStyles-button-7 MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            test-2
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedVertical MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedVertical MuiButtonGroup-groupedOutlinedPrimary makeStyles-button-7 MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            test-3
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-iFrameContainer-3"
  >
    <div
      className="makeStyles-root-10"
      data-testid="iframe"
    >
      <iframe
        allow="encryted-media"
        allowFullScreen={true}
        height="100%"
        src="test-url"
        title="title"
        width="100%"
      />
    </div>
  </div>
  <div />
</div>
`;
